src/common/heap.h

```diff
@@ -7,6 +7,9 @@ extern "C" {
 uint32_t heap_total_size;
 uint32_t heap_bytes_remaining;
 
+extern const char *heap_base;
+extern const char *heap_limit;
```

Comment:
I am not sure I like this API design. It exposes some internals of the heap which would be nicer to keep private.

What about adding a simple function like `bool mem_is_heap_allocated(void *ptr)`? (currently, it would have the super simple implementation, where it would only check whether `ptr` is within the heap's bounds)

Reply:
I don't really understand why this should be kept internal, but OK, no problem.

Reply:
The `mem_is_heap_allocated` gives user „less power“ and keeps things better encapsulated. 

For example, I can imagine that in future, we might want to have two memory blocks reserved for heap. In such case, we would just reflect that in the `mem_is_heap_allocated` implementation. With two exposed variables `heap_base` and `heap_limit`, we would have to refactor all its usages, which might get much more messy.

Reply:
Removed in favor of `mem_is_heap_allocated`.